File: 23_display_matrix.c
//C program to display matrix

#include <stdio.h>

int main(){
    
    int rows, cols;
    int num[rows][cols];//declared matrix
    
    printf("Enter number of rows: ");
    scanf("%d", &rows);
    printf("\nEnter number of columns: ");
    scanf("%d", &cols);
    
    printf("\nEnter elements of array: " );
    for(int i=0; i<rows; i++){
        for(int j=0; j<cols; j++){
        printf("\nEnter element at position (%d,%d): ", i+1, j+1);
        scanf("%d", &num[i][j]);
        }
    }
    
    //Display Matrix
    printf("\nDisplaying entered matrix: \n");
    for(int i=0; i<rows; i++){
        for(int j=0; j<cols; j++){
            printf("%d\t", num[i][j]);
        }
        printf("\n");
    }
    
    
    return 0;
}

File: 23_display_matrix.py
#python program to display a matrix

#design user-defined function
def display_matrix(matrix):
    """Display a matrix."""
    for row in matrix:
        for element in row:
            print(element, end=" ")
        print()  # Move to the next line for the next row

def main():
    # Get the dimensions of the matrix from the user
    rows = int(input("Enter the number of rows: "))
    cols = int(input("Enter the number of columns: "))

    # Initialize an empty matrix
    matrix = []

    # Get the elements of the matrix from the user
    for i in range(rows):
        row = []
        for j in range(cols):
            element = int(input(f"Enter element at position ({i + 1}, {j + 1}): "))
            row.append(element)
        matrix.append(row)

    # Display the matrix
    print("\nEntered matrix is: ")
    display_matrix(matrix)

if __name__ == "__main__":
    main()
